src/shogun/mathematics/linalg/internal/implementation/Variance.h

```diff
@@ -0,0 +1,236 @@
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation are those
+ * of the authors and should not be interpreted as representing official policies,
+ * either expressed or implied, of the Shogun Development Team.
+ */
+
+#ifndef SRC_SHOGUN_MATHEMATICS_LINALG_INTERNAL_IMPLEMENTATION_VARIANCE_H_
+#define SRC_SHOGUN_MATHEMATICS_LINALG_INTERNAL_IMPLEMENTATION_VARIANCE_H_
+
+#include <shogun/lib/config.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/lib/SGMatrix.h>
+//#include <shogun/mathematics/linalg/internal/implementation/MeanEigen3.h> To be included when matrix mean is implemented
+#include <shogun/mathematics/Statistics.h>
+#include <shogun/mathematics/eigen3.h>
+
+using namespace shogun::linalg;
+
+namespace shogun
+{
+
+namespace linalg
+{
+
+namespace implementation
+{
+
+
+/**
+ * @brief Generic class variance which provides a static compute method. This class
+ * is specialized for different types of vectors.
+ */
+template <enum Backend, class Vector>
+struct variance
+{
+	/** Scalar type */
+	typedef typename Vector::Scalar T;
+
+	/** Calculates unbiased empirical variance estimator of given values. Given
+	 * \f$\{x_1, ..., x_m\}\f$, this is
+	 * \f$\frac{1}{m-1}\sum_{i=1}^m (x-\bar{x})^2\f$ where
+	 * \f$\bar x=\frac{1}{m}\sum_{i=1}^m x_i\f$
+	 *
+	 * @param a vector of values
+	 * @return variance of given values
+	 */
+	static T compute(Vector a);
+};
+
+/**
+ * @brief Specialization of generic variance which works with SGVector
+ * and uses Eigen3 as backend for computing variance.
+ */
+template <class Vector>
+struct variance<Backend::EIGEN3, Vector>
+{
+	/** Scalar type */
+	typedef typename Vector::Scalar T;
+
+	/** Calculates unbiased empirical variance estimator of given values. Given
+	 * \f$\{x_1, ..., x_m\}\f$, this is
+	 * \f$\frac{1}{m-1}\sum_{i=1}^m (x-\bar{x})^2\f$ where
+	 * \f$\bar x=\frac{1}{m}\sum_{i=1}^m x_i\f$
+	 *
+	 * @param a vector of values
```

Comment:
The comments above regarding the method documentation also apply here and below.